lib/_http_client.js

```diff
@@ -636,7 +636,11 @@ ClientRequest.prototype.onSocket = function onSocket(socket) {
 function onSocketNT(req, socket) {
   if (req.aborted) {
     // If we were aborted while waiting for a socket, skip the whole thing.
-    socket.emit('free');
+    if (req.socketPath || !req.agent) {
+      socket.destroy();
+    } else {
+      socket.emit('free');
+    }
```

Comment:
I've added a test which should clarify why `socket.emit('free')` is used. It's for queued requests which are already aborted.